site/index.html

```diff
@@ -0,0 +1,152 @@
+<!doctype html>
+<html>
+	<head>
+		<title>goioi</title>
+
+		<style>
+body {
+	margin: 0;
+	background-color: #00cccc;
+}
+
+* {
+	box-sizing: border-box;
+}
+
+#dropzone {
+	width: calc(100vh - 100px);
+	height: calc(100vh - 100px);
+	margin-left: auto;
+	margin-right: auto;
+	margin-top: 50px;
+
+	background-color: #09dcdc;
+
+	text-align: center;
+	color: white;
+	font-size: 4em;
+	line-height: calc(100vh - 100px);
+	font-family: sans-serif;
+	font-weight: 100;
+
+	border-radius: 50%;
+	border: 3px white solid;
+
+	transition: .5s all cubic-bezier(0, 1, 0.2, 1);
+	transition-property: transform, width, border-radius, height, margin, line-height, background-color;
+	will-change: transform;
+}
+body.small #dropzone {
+	line-height: calc(100vw - 100px);
+	width: calc(100vw - 100px);
+	height: calc(100vw - 100px);
+}
+#dropzone.hovering {
+	transform: scale(1.1);
+}
+#dropzone.result {
+	width: 100vw !important;
+	height: 100vh !important;
+	margin-top: 0;
+
+	background-color: #02c598;
+
+	font-size: 2.3em;
+	font-family: 'molokai', 'monokai', consolas, monospace;
+	border-radius: 0;
+}
+#dropzone.result.error {
+	background-color: #a7013e;
+}
+		</style>
+	</head>
+
+	<body>
+		<div id="dropzone">
+			gooi a file
+		</div>
+		<script>
+function setWidthClass () {
+	document.body.className =
+		window.innerWidth < window.innerHeight ?
+			'small' :
+			'';
+}
+setWidthClass();
+window.addEventListener('resize', setWidthClass);
+
+function isFile (e) {
+	var dt = e.dataTransfer;
+
+	for (var i = 0; i < dt.types.length; i++) {
+		if (dt.types[i] === 'Files') {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+var dropzone = document.getElementById('dropzone');
+dropzone.addEventListener('dragenter', function (e) {
+	if (!isFile(e)) {
+		return;
+	}
+
+	dropzone.className = 'hovering';
+	dropzone.innerText = 'drop it';
+});
+dropzone.addEventListener('dragleave', function (e) {
+	e.preventDefault();
+	dropzone.className = '';
+	dropzone.innerText = 'gooi a file';
+});
+dropzone.addEventListener('dragover', function (e) {
+	e.preventDefault();
+});
+
+function showUrl (url) {
+	dropzone.className = 'result';
+	dropzone.innerText = url;
+}
+
+function showError (message) {
+	dropzone.className = 'result error';
+	dropzone.innerText = message;
+}
+
+dropzone.addEventListener('drop', function (e) {
+	e.preventDefault();
+
+	var file = e.dataTransfer.files[0];
+	var reader = new FileReader();
+	reader.onload = function () {
+		var arrbuf = reader.result;
+
+		var xhr = new XMLHttpRequest();
+
+		xhr.onreadystatechange = function (e) {
+			if (xhr.readyState !== 4) {
+				return;
+			}
+
+			if (xhr.status === 200) {
```

Comment:
Oh wait now I realise I didn't even check if the server returns 200 or something like a created status code.